services/workflows-service/src/collection-flow/controllers/collection-flow.controller.ts

```diff
@@ -154,7 +155,15 @@ export class CollectionFlowController {
 
       const directors = await Promise.all(
         workflowRuntimeData.context.entity.data.additionalInfo.directors?.map(
-          async (director: { firstName: string; lastName: string; email: string }) => {
+          async (director: { id?: string; firstName: string; lastName: string; email: string }) => {
+            // If ID is present then entity been created in KYB
+            if (director.id) {
```

Comment:
You're certain id is correct?
```suggestion
            if (director.ballerineEntityId) {
```

Reply:
Yes, `id` is recieved from creation endpoint and being written to entity in context.